packages/toolkit/src/query/core/buildMiddleware/invalidationByTags.ts

```diff
@@ -55,12 +71,27 @@ export const buildInvalidationByTagsHandler: InternalHandlerBuilder = ({
   }
 
   function invalidateTags(
-    tags: readonly FullTagDescription<string>[],
+    newTags: readonly FullTagDescription<string>[],
     mwApi: SubMiddlewareApi
   ) {
     const rootState = mwApi.getState()
     const state = rootState[reducerPath]
 
+    pendingTagInvalidations.push(...newTags)
```

Comment:
:hand: This looks like we could end up with the same tags showing up in `pendingTagInvalidations` multiple times, which I think also means we'd end up running some extra logic when we try to invalidate.  Can `pendingTagInvalidations` be a `Set` instead?

Reply:
No, since tags are objects and not strings in general. More specifically, their type is 

```ts
{
    type: TagType;
    id?: string | number | undefined;
}
```

The set of invalidated endpoints is already deduplicated in [`selectInvalidatedBy`](https://github.com/GeorchW/redux-toolkit/blob/cfa9fe3c46affd21aef996ce10ec1cf61a76603a/packages/toolkit/src/query/core/buildSelectors.ts#L209). Now, we could try de-duplicating here again by converting these tag objects into canonical string forms, but I think this is creating more overhead than it alleviates except in extreme cases.

Reply:
Oh, and while its true that tags can end up in `pendingTagInvalidations` multiple times, the amount of processing required is no different than if they were invalidated immediately. So this is not a performance regression. (It's not like this code will show up in the profiler anyway.)